File: cesta/AMarotoChivite-cesta/AMarotoChivite-cesta-Java/src/test/java/simulacionCesta/factories/CestListFactoryTest.java
package simulacionCesta.factories;

import org.junit.jupiter.api.Test;
import simulacionCesta.models.CestList;
import simulacionCesta.models.Product;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class CestListFactoryTest {

    @Test
    public void cestListFactoryCreateTest() {
        CestList cestList = CestListFactory.create();
        // Comprobar que la longitud de la lista de la cesta es 10
        assertEquals(10, cestList.getStorageProducts().length);
    }

    @Test
    public void checkProductNoRepeatTest() {
        CestList cestList = CestListFactory.create();
        Product[] productsStorage = cestList.getStorageProducts();
        boolean noRepeat = true;
        for (int i = 0; i < productsStorage.length; i++) {
            for (int j = i + 1; j < productsStorage.length; j++) {
                if (productsStorage[i].name.equals(productsStorage[j].name)) {
                    noRepeat = false;
                    break;
                }
            }
        }
        assertTrue(noRepeat);
    }
}

File: cesta/AMarotoChivite-cesta/AMarotoChivite-cesta-Kotlin/src/test/kotlin/simulacionCesta/factories/CestListFactoryTest.kt
package simulacionCesta.factories

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class CestListFactoryTest {
    @Test
    fun cestListFactoryCreateTest() {
        val cestList = CestListFactory.create()
        // Comprobar que la longitud de la lista de la cesta es 10
        Assertions.assertEquals(10, cestList.storageProducts.size)
    }

    @Test
    fun checkProductNoRepeatTest() {
        val cestList = CestListFactory.create()
        val productsStorage = cestList.storageProducts
        var noRepeat = true
        for (i in productsStorage.indices) {
            for (j in i + 1 until productsStorage.size) {
                if (productsStorage[i]!!.name == productsStorage[j]!!.name) {
                    noRepeat = false
                    break
                }
            }
        }
        Assertions.assertTrue(noRepeat)
    }
}